src/core/main.c

```diff
@@ -1532,7 +1532,8 @@ int main(int argc, char *argv[]) {
                  * need to do that for user instances since they never log
                  * into the console. */
                 log_show_color(colors_enabled());
-                make_null_stdio();
+                if (make_null_stdio() < 0)
+                        log_warning_errno(errno, "Failed to redirect standard streams to /dev/null: %m");
```

Comment:
in systemd, all functions return negative errno values for indicating errors. Thus, unless the function you call is part of the glibc API you shouldn't check and use "errno", but the actual return value. Hence, please write this as:

```c
r = make_null_stdio();
if (r < 0) 
        log_warning_errno(r, "Failed …");
```